include/zephyr/modem/backend/stat.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/kernel.h>
+
+#ifndef ZEPHYR_MODEM_PIPESTAT_
+#define ZEPHYR_MODEM_PIPESTAT_
+
+/**
+ * @cond INTERNAL_HIDDEN
+ */
+
+/** Modem backend statistics instance structure */
+struct modem_backend_stat {
+	sys_snode_t node;
+	char name[16];
+	uint32_t transmit_buf_max;
+	uint32_t transmit_buf_capacity;
+	uint32_t receive_buf_max;
+	uint32_t receive_buf_capacity;
+};
+
+/**
+ * @endcond
+ */
+
+/** Modem backend statistics advertisement structure */
+struct modem_backend_stat_adv {
+	/** Number of bytes in transmit buffer, 0 if not implemented */
+	uint32_t transmit_buf_size;
+	/** Number of bytes in receive buffer, 0 if not implemented */
+	uint32_t receive_buf_size;
+};
+
+/** Modem backend statistics configuration structure */
+struct modem_backend_stat_config {
+	/** Name assigned to modem backend statistics instance */
+	const char *name;
+	/** Total capacity of transmit buffer, 0 if not implemented */
+	uint32_t transmit_buf_capacity;
+	/** Total capacity of receive buffer, 0 if not implemented */
+	uint32_t receive_buf_capacity;
+};
+
+/**
+ * @brief Initialize modem backend statistics instance
+ *
+ * @param stat Modem backend statistics instance
+ * @param config Modem backend statistics instance configuration
+ *
+ */
+void modem_backend_stat_init(struct modem_backend_stat *stat,
+			     const struct modem_backend_stat_config *config);
+
+/**
+ * @brief Advertise modem backend statistics
+ *
+ * @param stat Modem backend statistics instance
+ * @param adv Advertisement
+ *
+ * @note Invoke when either buffer decreases in size
+ * @note Safe to invoke from ISR
+ */
+void modem_backend_stat_advertise(struct modem_backend_stat *stat,
```

Comment:
As this function is always used to update only one field at a time (either transmit or receive buffer length) why not make it so that you just pass an `uint32_t` (along with a field identifier) instead of a `struct modem_backend_stat_adv`? This would reduce the number of lines needed for callers of this.

Reply:
Strategy changed to match :)